sdk/trace/src/main/java/io/opentelemetry/sdk/trace/SdkSpanBuilder.java

```diff
@@ -234,6 +227,14 @@ public Span startSpan() {
         startEpochNanos);
   }
 
+  private AttributesMap attributes() {
+    AttributesMap attributes = this.attributes;
+    if (attributes == null) {
+      attributes = this.attributes = new AttributesMap(spanLimits.getMaxNumberOfAttributes());
```

Comment:
hm. I'd definitely prefer to avoid the double assignment in a single line, just from a readability perspective.